dockertest/images.py

```diff
@@ -54,11 +54,11 @@ class DockerImage(object):  # pylint: disable=R0902
     #: parsing, spec defined in docker-io documentation.  e.g.
     #: ``[registry_hostname[:port]/][user_name/]
     #: (repository_name[:version_tag])``
-    repo_split_p = re.compile(r"(?P<repo_addr>.+?"  # optional
-                              r"(?P<repo_addr_port>:[\d]+?)?/)?"  # optional
-                              r"(?P<user>[\w\-\.\+]+/)?"  # optional
-                              r"(?P<repo>[\w\-\.]+)"  # required
-                              r"(?P<tag>:[\w\-\.]+)?")  # optional
+    repo_split_p = (r"(?P<repo_addr>.+?"  # optional
+                    r"(?P<repo_addr_port>:[\d]+?)?/)?"  # optional
+                    r"(?P<user>[\w\-\.\+]+/)?"  # optional
+                    r"(?P<repo>[\w\-\.]+)"  # required
+                    r"(?P<tag>:[\w\-\.]*)?")  # optional
```

Comment:
Note also that the '+' in the last (tag) line is now '*', because "foo/bar:" (empty tag) seems to be valid syntax.